jupyterlab_gcedetails/src/components/hardware_scaling_status.spec.tsx

```diff
@@ -0,0 +1,218 @@
+    mockStopInstance.mockResolvedValue(mockInstance);
+    mockSetMachineType.mockResolvedValue(mockInstance);
+    mockSetAccelerator.mockResolvedValue(mockInstance);
+    mockStartInstance.mockResolvedValue(mockInstance);
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const resolveValue = Promise.resolve({ ...detailsResponse });
+    mockGetUtilizationData.mockReturnValue(resolveValue);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    expect(hardwareScalingStatus.state('status')).toEqual(Status.Authorizing);
+    expect(hardwareScalingStatus).toMatchSnapshot('Authorizing');
+    expect(mockAuthTokenRetrieval).toBeCalled();
+    await mockAuthTokenRetrieval();
+    expect(mockNotebookService.setAuthToken).toHaveBeenCalledWith(mockToken);
+    expect(hardwareScalingStatus.state('status')).toEqual(
+      Status['Stopping Instance']
+    );
+    expect(hardwareScalingStatus).toMatchSnapshot('Stopping Instance');
+    expect(mockNotebookService.stop).toBeCalled();
+    await mockStopInstance();
+    expect(hardwareScalingStatus.state('status')).toEqual(
+      Status['Updating Machine Configuration']
+    );
+    expect(hardwareScalingStatus).toMatchSnapshot(
+      'Updating Machine Configuration'
+    );
+    expect(mockNotebookService.setMachineType).toBeCalled();
+    await mockSetMachineType();
+    expect(hardwareScalingStatus.state('status')).toEqual(
+      Status['Updating GPU Configuration']
+    );
+    expect(hardwareScalingStatus).toMatchSnapshot('Updating GPU Configuration');
+    expect(mockNotebookService.setAccelerator).toBeCalled();
+    await mockSetAccelerator();
+    expect(hardwareScalingStatus.state('status')).toEqual(
+      Status['Starting Instance']
+    );
+    expect(hardwareScalingStatus).toMatchSnapshot('Starting Instance');
+    expect(mockNotebookService.start).toBeCalled();
+    await mockStartInstance();
+    expect(hardwareScalingStatus.state('status')).toEqual(
+      Status['Refreshing Session']
+    );
+    expect(hardwareScalingStatus).toMatchSnapshot('Refreshing Session');
+    await mockGetUtilizationData();
+    setTimeout(() => {
+      expect(mockOnComplete).toBeCalled();
+      expect(hardwareScalingStatus.state('status')).toEqual(Status.Complete);
+      expect(hardwareScalingStatus).toMatchSnapshot('Complete');
+      hardwareScalingStatus
+        .find(Button)
+        .first()
+        .simulate('click');
+      expect(mockOnDialogClose).toBeCalled();
+    }, 1000);
+  });
+
+  it('Renders with auth error', async () => {
+    mockAuthTokenRetrieval.mockImplementation(() => {
```

Comment:
I think it's more customary and simpler to use mockRejectedvalue or mockReturnValue with a rejected promise.

https://jestjs.io/docs/en/mock-function-api#mockfnmockrejectedvaluevalue